test/loader_tests.js

```diff
@@ -150,6 +150,48 @@ describe('component loader', () => {
     assert(false, 'Expected an exception');
   });
 
+  it('should fail when a sync setup function fails', async () => {
+    let load = subject({
+      fail: {
+        requires: [],
+        setup: () => {
+          throw new Error('uhoh!');
+        },
+      },
+    });
+
+    try {
+      await load('fail');
+    } catch (e) {
+      if (!e.message.match(/uhoh!/)) {
+        throw e;
+      }
+      return; // Ignore expected error
+    }
+    assert(false, 'Expected an exception');
+  });
+
+  it('should fail when an async setup function fails', async () => {
+    let load = subject({
+      fail: {
+        requires: [],
+        setup: async () => {
+          throw new Error('uhoh!');
```

Comment:
We'd need another case here for a promise returning function, which is a valid way to write a setup function.

Example:
```diff
diff --git a/test/loader_tests.js b/test/loader_tests.js
index 660ba02..192416c 100644
--- a/test/loader_tests.js
+++ b/test/loader_tests.js
@@ -175,8 +175,8 @@ describe('component loader', () => {
     let load = subject({
       fail: {
         requires: [],
-        setup: async () => {
-          throw new Error('uhoh!');
+        setup: () => {
+          return Promise.reject('uhoh!');
         },
       },
     });
```

causes the test to fail with:

```
     TypeError: Cannot read property 'match' of undefined
      at _callee12$ (.test/loader_tests.js:440:39)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:65:40)
      at Generator.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:299:22)
      at Generator.prototype.(anonymous function) [as throw] (node_modules/regenerator-runtime/runtime.js:117:21)
      at step (node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
      at node_modules/babel-runtime/helpers/asyncToGenerator.js:30:13
      at process._tickCallback (internal/process/next_tick.js:68:7)
```

which is the line:
```javascript
            if (_context12.t0.message.match(/uhoh!/)) {
              _context12.next = 10;
              break;
            }
```

This is because the return value in this case is a promise, not the rejection reason of a promise, so it should have a `.then()` instead of a `.message` property.

Reply:
nvm this, I did it wrong, the problem is that i'm returning directly the string and not the error